### PR TITLE
Feature/ui

### PR DESCRIPTION
**Pull request by Tanav1**

### CODE REVIEW

**Comment by Tanav1** [comment]
Update to donors which included the change from forms to modelForms and other code cleanup.

File: src/donors/models.py [hunk 1]
```diff
@@ -38,3 +39,51 @@ class RecurrenceChoices(models.TextChoices):
     def __str__(self):
         return f"{self.get_recurrence_display()} donation from {self.donor.name}"
 
+class tempEmergencyContact(models.Model):
```
Comment (askvortsov1) [inline src/donors/models.py:42]: What are these temp classes, and why are they camelCased? Class names are always PascalCased in python.

File: src/donors/models.py [hunk 2]
```diff
@@ -38,3 +39,51 @@ class RecurrenceChoices(models.TextChoices):
...
+    volunteer_name = models.CharField(max_length=200)
```
Comment (askvortsov1) [inline src/donors/models.py:43]: Why are these directly stored as strings, and not as foreignkey references to users?

File: src/donors/models.py [hunk 3]
```diff
@@ -38,3 +39,51 @@ class RecurrenceChoices(models.TextChoices):
...
+    second_phone_number = models.CharField(max_length=200)
+    
+    backup_contact = models.CharField(max_length=200)
+    backup_contact_relationship = models.CharField(max_length=200)
+    backup_first_phone = models.CharField(max_length=200)
+    backup_second_phone = models.CharField(max_length=200)
+
+
+class tempWaiver(models.Model):
+    volunteer_signature = models.CharField(max_length=200)
```
Comment (askvortsov1) [inline src/donors/models.py:56]: Which of this data actually needs to be stored/persisted for the future? How does storing a string with the volunteer's name help us here?

File: src/donors/models.py [hunk 4]
```diff
@@ -38,3 +39,51 @@ class RecurrenceChoices(models.TextChoices):
...
+    zip_code = models.CharField(max_length=200)
+    email = models.CharField(max_length=200)
+
+    group = models.CharField(max_length=200, blank = True)
+
+    interest = models.CharField(max_length=200, default = "Packing, Client pantry, etc.", blank = True)
+
+
+    pennsylvania_child_abuse_history_certification = models.BooleanField(blank = True) 
+    date_of_certification1 = models.CharField(max_length=200, default= "DD/MM/YYYY", blank = True)
```
Comment (askvortsov1) [inline src/donors/models.py:75]: There is a models.datefield. Using charfield for everything is bad practice.

File: src/donors/models.py [hunk 5]
```diff
@@ -38,3 +39,51 @@ class RecurrenceChoices(models.TextChoices):
...
+    backup_second_phone = models.CharField(max_length=200)
+
+
+class tempWaiver(models.Model):
+    volunteer_signature = models.CharField(max_length=200)
+    date = models.CharField(max_length=200,default= "DD/MM/YYYY")
+    parents_signature = models.CharField(max_length=200, blank = True)
+    
+class tempVolunteerSignup(models.Model):
+    volunteer_name = models.CharField(max_length=200)
```
Comment (askvortsov1) [inline src/donors/models.py:61]: Why are all of these charfields with 200 characters length? Good database design is important. If I ask you, you should be able to provide a convincing argument for EVERY field on here, why it is the type it is, and why it has the settings it does. Please revisit ALL of these and implement them properly with proper data types, relations, etc

File: src/donors/models.py [hunk 6]
```diff
@@ -38,3 +39,51 @@ class RecurrenceChoices(models.TextChoices):
...
+    phone_number = models.CharField(max_length=200)
+    address = models.CharField(max_length=200)
+    city = models.CharField(max_length=200)
+    state = models.CharField(max_length=200)
+    zip_code = models.CharField(max_length=200)
+    email = models.CharField(max_length=200)
+
+    group = models.CharField(max_length=200, blank = True)
```
Comment (askvortsov1) [inline src/donors/models.py:69]: What is this? What does group mean?

File: src/donors/models.py [hunk 7]
```diff
@@ -38,3 +39,51 @@ class RecurrenceChoices(models.TextChoices):
...
+
+    interest = models.CharField(max_length=200, default = "Packing, Client pantry, etc.", blank = True)
```
Comment (askvortsov1) [inline src/donors/models.py:71]: Why is this 200 characters?

File: src/donors/models.py [hunk 8]
```diff
@@ -38,3 +39,51 @@ class RecurrenceChoices(models.TextChoices):
...
+
+
+    pennsylvania_child_abuse_history_certification = models.BooleanField(blank = True)
```
Comment (askvortsov1) [inline src/donors/models.py:74]: Shouldn't we collect documentation for these?

File: src/donors/models.py [hunk 9]
```diff
@@ -38,3 +39,51 @@ class RecurrenceChoices(models.TextChoices):
...
+
+    pennsylvania_state_police_criminal_record_check = models.BooleanField(blank = True)
+    date_of_certification2 = models.CharField(max_length=200, default= "DD/MM/YYYY", blank = True)
+
+
+    fbi_federal_criminal_history_check = models.BooleanField(blank = True)
+    date_of_certification3 = models.CharField(max_length=200, default= "DD/MM/YYYY", blank = True)
+  
+    
+    volunteer_signature = models.CharField(max_length=200)
```
Comment (askvortsov1) [inline src/donors/models.py:85]: Once again, always ask yourself why you are doing what you are doing. The purpose of databases is to persist data across requests. We care when the form was signed/submitted. But since this isn't an actual signature, we gain nothing from storing it. This would be like storing recaptcha submissions to the database.

File: src/donors/models.py [hunk 10]
```diff
@@ -38,3 +39,51 @@ class RecurrenceChoices(models.TextChoices):
...
+
+    pennsylvania_child_abuse_history_certification = models.BooleanField(blank = True) 
+    date_of_certification1 = models.CharField(max_length=200, default= "DD/MM/YYYY", blank = True) 
+
+    pennsylvania_state_police_criminal_record_check = models.BooleanField(blank = True)
+    date_of_certification2 = models.CharField(max_length=200, default= "DD/MM/YYYY", blank = True)
+
+
+    fbi_federal_criminal_history_check = models.BooleanField(blank = True)
+    date_of_certification3 = models.CharField(max_length=200, default= "DD/MM/YYYY", blank = True)
```
Comment (askvortsov1) [inline src/donors/models.py:82]: what is certification3? If I look at the database without the context of this model, will I know what this column means?

File: src/donors/templates/emergencycontact.html [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+{% block title %} Emergency Contact {% endblock %}
+
+{% block content%}
+<form method = "post" action = submission/>
+  
+  
+  {% csrf_token %}
+  <h3>Emergency Contact Form</h3>
+  
+  {{form.as_p}}
```
Comment (askvortsov1) [inline src/donors/templates/emergencycontact.html:11]: Crispy forms, please.

File: src/donors/templates/emergencycontact.html [hunk 12]
```diff
@@ -0,0 +1,17 @@
+{% extends 'base.html' %}
+{% block title %} Emergency Contact {% endblock %}
+
+{% block content%}
+<form method = "post" action = submission/>
+  
+  
+  {% csrf_token %}
+  <h3>Emergency Contact Form</h3>
```
Comment (askvortsov1) [inline src/donors/templates/emergencycontact.html:9]: The header should not be in the form.

File: src/donors/templates/submission.html [hunk 13]
```diff
@@ -0,0 +1,11 @@
+<h1>Thanks for submitting!</h1>
+
+
+{% csrf_token %}
```
Comment (askvortsov1) [inline src/donors/templates/submission.html:4]: Why is this here? What does this do?

File: src/donors/templates/submission.html [hunk 14]
```diff
@@ -0,0 +1,11 @@
...
+
+<form action="/donors/" >
```
Comment (askvortsov1) [inline src/donors/templates/submission.html:6]: This is unnecessary. We should just redirect them back to the proper page after they submit, and use django's messages system to show a success message.

File: src/donors/templates/submit.php [hunk 15]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+
+//Opens new file to write
+$data_file = fopen("example.txt", "w");
+
+$name = $_POST["name"];
```
Comment (askvortsov1) [inline src/donors/templates/submit.php:7]: Why is there PHP here at all? This shouldn't work, and shouldn't be here.

File: src/donors/templates/volunteersignup.html [hunk 16]
```diff
@@ -0,0 +1,28 @@
+{% extends 'base.html' %}
+{% block title %} Volunteer Signup {% endblock %}
+
+{% block content%}
+<form method = "post" action = submission/ >
```
Comment (askvortsov1) [inline src/donors/templates/volunteersignup.html:5]: post should be all caps. For action, it should be in quotes and use django's dynamic URL generator (NEVER hardcode url paths).

File: src/donors/templates/volunteersignup.html [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+<h3>Volunteer Signup Application</h3>
+{% csrf_token %}
+
+
+{{form.as_p}}
+
+<h3>Your signature indicates that you have read the volunteer manual and agree to
```
Comment (askvortsov1) [inline src/donors/templates/volunteersignup.html:12]: Why is this text in header tags? It is not functionally a header; if you want to make it bigger, css should be used?

File: src/donors/templates/waiver.html [hunk 18]
```diff
@@ -0,0 +1,124 @@
...
+    Cocoa Packs volunteer.
+    </div>
+    
+    <div>
+    • You authorize Cocoa Packs to provide you first aid, emergency
+    medical assistance, and transportation.
+    </div>
+    
+    <div>
+    • You agree to hold in confidence at all times, during and after
```
Comment (askvortsov1) [inline src/donors/templates/waiver.html:44]: These should be wrapped in paragraph tags, not div tags.

File: src/donors/urls.py [hunk 19]
```diff
@@ -1,8 +1,15 @@
 from django.urls import path
 from . import views
+from django.conf.urls import url
 
 app_name = "donors"
 
 urlpatterns = [
-    path('', views.home, name='home')
-]
+    path('', views.home, name=''),
```
Comment (askvortsov1) [inline src/donors/urls.py:8]: These all need names. Also, url is deprecated and should NOT be used

File: src/donors/models.py (hunk 6 above)
Comment (Tanav1) [reply src/donors/models.py:69]: This field is for if you have a preference

**Comment by askvortsov1** [comment]
@Tanav1 why was this PR closed?
